<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConfigDialog</class>
 <widget class="QDialog" name="ConfigDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>405</width>
    <height>243</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="general">
        <attribute name="title">
         <string notr="true">General</string>
        </attribute>
        <widget class="QWidget" name="verticalLayoutWidget">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>242</width>
           <height>103</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0">
          <property name="sizeConstraint">
           <enum>QLayout::SetMinimumSize</enum>
          </property>
          <item>
           <layout class="QFormLayout" name="formLayout">
            <item row="2" column="0">
             <widget class="QLabel" name="label">
              <property name="whatsThis">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Only used, when &amp;quot;RenderWholeSong&amp;quot; is active:&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Determines, how many milliseconds of a song shall be prepared (i.e. rendered), before playback can actually start.&lt;/p&gt;&lt;p&gt;Value of 0: Playback will start after the whole song has been rendered.&lt;/p&gt;&lt;p&gt;Value of 1: Playback will start nearly immediately, i.e. after 1 ms of sound has been generated. This can result in a few hearable cracks, due to unfinished sound (esp. true for emulated sound formats).&lt;/p&gt;&lt;p&gt;Bigger values ensure that more sound has been prepared ready for playback. However they will also &lt;a name=&quot;result_box&quot;/&gt;delay the start of playing back the next song.&lt;/p&gt;&lt;p&gt;500 ms is usually a good compromise.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>PreRenderTime [ms]</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="spinPreRenderTime">
              <property name="maximum">
               <number>9999</number>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_2">
              <property name="whatsThis">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Render the whole Song at once?&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Yes: Needs a lot of memory. But makes seeking possible&lt;/p&gt;&lt;p&gt;No: Little memory, no seeking though.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;Note:&lt;/span&gt; This option is a suggestion. If there is not enough memory available&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>RenderWholeSong</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QCheckBox" name="checkRenderWhole">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_3">
              <property name="whatsThis">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Use the audio normalization data generated by anmp-normalize?&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>useAudioNormalization</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QCheckBox" name="checkAudioNorm">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="comboBoxAudioDriver"/>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_13">
              <property name="whatsThis">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Specifies the audio API ANMP should use.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>Audio Driver</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="howToPlay">
        <attribute name="title">
         <string notr="true">How-To-Play</string>
        </attribute>
        <widget class="QWidget" name="formLayoutWidget">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>236</width>
           <height>136</height>
          </rect>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Use Loop Info</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QCheckBox" name="checkLoopInfo">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Overriding Loop Count</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="spinLoopCount">
            <property name="minimum">
             <number>-1</number>
            </property>
            <property name="maximum">
             <number>999</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Fade Time Stop</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="spinFadeStop">
            <property name="maximum">
             <number>120000</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Fade Time Pause</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="spinFadePause">
            <property name="maximum">
             <number>120000</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="usf">
        <attribute name="title">
         <string notr="true">USF</string>
        </attribute>
        <widget class="QWidget" name="formLayoutWidget_2">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>163</width>
           <height>80</height>
          </rect>
         </property>
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="label_8">
            <property name="whatsThis">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Whether to use High-Level-Emulation (HLE) or not (i.e. Low-Level-Emulation (LLE)). Unless you are audiophil, there is no reason to set this to false, since HLE became pretty accurate over the last years. However you should definitly set this to true when being on a mobile device.&lt;/p&gt;&lt;p&gt;True: Speeds up emulation (about factor 2 or more) at the cost of accuracy and potential emulation bugs.&lt;/p&gt;&lt;p&gt;False: Busy CPU, but most accurate emulation.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Enable HLE</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QCheckBox" name="usfUseHle">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="gme">
        <attribute name="title">
         <string notr="true">GME</string>
        </attribute>
        <widget class="QWidget" name="formLayoutWidget_3">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>215</width>
           <height>101</height>
          </rect>
         </property>
         <layout class="QFormLayout" name="formLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Sample Rate</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="spinGmeSampleRate">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>96000</number>
            </property>
            <property name="singleStep">
             <number>1000</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_10">
            <property name="whatsThis">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Adjust stereo echo depth, where 0.0 = off and 1.0 = maximum.&lt;/p&gt;&lt;p&gt;Has no effect for GYM, SPC, and Sega Genesis VGM music&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Echo Depth</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="spinGmeSfx">
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_11">
            <property name="whatsThis">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;IMHO, there is no reason to turn this off. CPU costs are not worth it either.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Accurate Emulation</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="checkGmeAccurate">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Play Forever</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QCheckBox" name="checkGmeForever">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ConfigDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ConfigDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
